hydra-access-controls/app/models/hydra/access_controls/permission.rb

```diff
@@ -71,16 +83,24 @@ def build_agent(name, type)
       raise "Can't build agent #{inspect}" unless name && type
       self.agent = case type
                    when 'group'
-                     build_agent_resource(GROUP_AGENT_URL_PREFIX, name)
+                     build_agent_resource(GROUP_AGENT_PATH, name)
                    when 'person'
-                     build_agent_resource(PERSON_AGENT_URL_PREFIX, name)
+                     build_agent_resource(PERSON_AGENT_PATH, name)
                    else
                      raise ArgumentError, "Unknown agent type #{type.inspect}"
                    end
     end
 
-    def build_agent_resource(prefix, name)
-      [Agent.new(::RDF::URI.new("#{prefix}##{URI.encode(name)}"))]
+    # The current URL.hash standard (As of March 2021) is that the post-hash portion of the URL is not percent-decoded
+    # however in order to ensure backward compatibility with already recorded values we are normalizing
+    # the fragment here. See https://developer.mozilla.org/en-US/docs/Web/API/URL/hash
+    def build_agent_rdf(path, name)
+      rdf_uri = ::RDF::URI.new(scheme: AGENT_URL_SCHEME, host: AGENT_URL_HOST, path: path, fragment: name)
+      ::RDF::URI.new(scheme: AGENT_URL_SCHEME, host: AGENT_URL_HOST, path: path, fragment:rdf_uri.normalized_fragment)
```

Comment:
Would it work to use `.normalize!`?
```suggestion
      ::RDF::URI.new(scheme: AGENT_URL_SCHEME, host: AGENT_URL_HOST, path: path, fragment: name).normalize!
```

Reply:
Oh! I tried that before I added the fragment part, when I was concatenating the path. Nice.